kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -235,6 +226,39 @@ public Optional<Predicate> getRemainingFilter() {
     return getDataFilters();
   }
 
+  private StructType getPhysicalSchema() {
```

Comment:
I think it would be useful to have docs here it's not obvious what this function does. I think it does the following essentially right?
1. convert logical fields to physical based on CM mode for non-metadata fields
2. convert metadata fields to physical fields if applicable (i.e. row tracking)
3. adds row_index if needed for DVs

Reply:
also nit - I think since this does real generation "get" would be better named "generate" or something